crates/config/src/inline/conf_parser.rs

```diff
@@ -41,10 +48,27 @@ where
     fn merge(&self, natspec: &NatSpec) -> Result<Option<Self>, InlineConfigError> {
         let config_key = Self::config_key();
 
-        let configs = natspec
-            .current_profile_configs()
-            .filter(|l| l.contains(&config_key))
-            .collect::<Vec<String>>();
+        let mut configs = Vec::new();
+        for line in natspec.current_profile_configs() {
+            if !VALID_CONFIG_KEYS.iter().any(|&key| line.contains(key)) {
+                let wrong_key = line
+                    .split('.')
+                    .nth(1) // Get the part after the first dot (profile.KEY.value)
+                    .unwrap_or("unknown-config-key")
+                    .to_string();
+                return Err(InlineConfigError {
+                    line,
+                    source: InlineConfigParserError::InvalidConfigKey(
+                        wrong_key,
+                        format!("{VALID_CONFIG_KEYS:?}"),
+                    ),
+                });
```

Comment:
I think a warning would be sufficient here given that the key is simply ignored when continuing